### DISCUSSION TITLE
Rendering Quarto book to pdf - Formatting issues

### DISCUSSION THREAD

**Discussion by faaa6** [quarto-dev/quarto-cli#4768] [Q&A]
### Bug description

I am currently working on a larger project utilising the strcture of multiple .qmd documents which I have to render to a pdf.
While doing this I have encountered some formatting issues that I can't seem to find solutions to.

1. TOC/LOF/LOT
I would like to include a table of content, list of figures and list of tables in my document and have specified these in my _quarto.yml document as follows:

format:
  pdf:
    documentclass: scrreprt
    links-as-notes: true
    reference-location: section
    toc: true
    toc-depth: 2
    lof: true
    lot: true
    number-sections: true
    fig-width: 8
    fig-height: 6

However in the final document I have pages devoted to these parts of the document, but these are empty and don't show the chapters, tables and figures of my document.

2. Footnotes

I would like to include footnotes at the end of each chapter (or even preferably at the end of the book in an own section) which does not seem to work in pdf, even though I have included reference-location: section in my .yml document.

3. Cross referencing

Cross referencing does not work across as well as within the different parts of the documents for chapters/sections, figures and tables.

**Comment by faaa6** [comment]
OK, I need to redact point 1. and 3.
There was an issue in the rendering due to my code that did halted the process before completion.
So only the footnote issue needs solving :)

**Reply by jdutant** [reply]
There's probably a LaTeX solution. [Try adding this](https://tex.stackexchange.com/questions/56145/is-there-a-way-to-move-all-footnotes-to-the-end-of-the-document) to your quarto options:

```yaml
format:
  pdf: 
    include-in-header: 
      text: |
          \usepackage{endnotes}
          \let\footnote=\endnote
```

And [add to the end of each chapter](https://tex.stackexchange.com/questions/9475/how-does-one-get-endnotes-at-the-end-of-every-chapter) the following line:

```
\theendnotes
```

Haven't tested it, but provided the LaTeX endnotes package is compatible with the KOMA class used by quarto to generate PDF documents that should work.

**Comment by faaa6** [comment]
This works quite nicely (removes the cross reference but this is currently not relevant to me).
Thank you very much :)
